Fix CPS/JTF hang in VsPackageInstaller

PatoBeltran opened the pull request:
## Bug
Fixes: VSTS 508282

## Fix
Make sure JTF is loaded with the correct context if CPS already running in VsPackageInstaller.

PatoBeltran committed:
Fix CPS/JTF hang in VsPackageInstaller

dnfclas commented:
@PatoBeltran,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

jainaashish commented on src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs:
```diff
@@ -53,7 +59,26 @@ public class VsPackageInstaller : IVsPackageInstaller2
             _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
 
-            PumpingJTF = new PumpingJTF(NuGetUIThreadHelper.JoinableTaskFactory.Context);
+            PumpingJTF = null;
```
You don't need to explicitly set it to null.

jainaashish commented on src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs:
```diff
@@ -53,7 +59,26 @@ public class VsPackageInstaller : IVsPackageInstaller2
+        {
+            if (PumpingJTF == null)
+            {
+                var VsHierarchy = VsHierarchyItem.FromDteProject(project).VsHierarchy;
```
Why do you need to create `VsHierarchyItem` here? can't you just directly use `VsHierarchyUtility.ToVsHierarchy(project)`

jainaashish commented on src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs:
```diff
@@ -53,7 +59,26 @@ public class VsPackageInstaller : IVsPackageInstaller2
+
+        private void RunJTFWithCorrectContext(Project project, Func<Task> asyncTask)
+        {
+            if (PumpingJTF == null)
```
You need to check for CPS capability for each project, these apis are being called. Since there could be a case where initial project was not CPS enabled but later on one more project was added which was CPS. But in your case, you only check first time.

jainaashish reviewed (changes requested):
Need to make sure we run this CPS capability check each time IVS api is being called or set some boolean to know if CPS jtf is already set or not.

PatoBeltran committed:
Check for CPS JTF until it is loaded then just use that

PatoBeltran commented:
@jainaashish addressed comments. If CPS is not loaded will check for each project, once CPS is loaded it will create a new JTF with the correct context and keep using that one.

PatoBeltran commented:
@jainaashish addressed comments. If CPS is not loaded will check for each project, once CPS is loaded it will create a new JTF with the correct context and keep using that one.

jainaashish reviewed (approved):
Looks good, make sure to run a private val build just to make sure there is no RPS regression for this.

PatoBeltran commented:
Merging... DDRITs and RPS are green.
